cg/meta/workflow/nf_analysis.py

```diff
@@ -308,26 +309,70 @@ def run_analysis(
             )
 
     @staticmethod
-    def get_deliverables_template_content() -> list[dict]:
+    def get_deliverables_template_content(self) -> list[dict]:
         """Return deliverables file template content."""
         raise NotImplementedError
 
-    def get_deliverables_for_case(self, case_id: str) -> PipelineDeliverables:
-        """Return PipelineDeliverables for a given case."""
-        deliverable_template: list[dict] = self.get_deliverables_template_content()
-        sample_id: str = self.status_db.get_samples_by_case_id(case_id).pop().internal_id
+    @staticmethod
+    def get_bundle_filenames_path(self) -> Path | None:
+        """Return bundle filenames path."""
+        return None
+
+    @staticmethod
+    def get_formatted_file_deliverable(
+        file_template: dict[str | None, str | None],
+        case_id: str,
+        sample_id: str,
+        sample_name: str,
+        case_path: str,
+    ) -> FileDeliverable:
+        """Return the formatted file deliverable with the case and sample attributes."""
+        deliverables = file_template.copy()
+        for deliverable_field, deliverable_value in file_template.items():
+            if deliverable_value is None:
+                continue
+            deliverables[deliverable_field] = (
+                deliverables[deliverable_field]
+                .replace("CASEID", case_id)
+                .replace("SAMPLEID", sample_id)
+                .replace("SAMPLENAME", sample_name)
+                .replace("PATHTOCASE", case_path)
+            )
+        return FileDeliverable(**deliverables)
```

Comment:
```suggestion
        return FileDeliverable.model_validate(deliverables)
```

Reply:
I will leave it as it is for now. `models/nf_analysis.py` use pydantic v.1 and I am thinking if this should be handled as a separate PR.